frontend/src/state/GraphState.jsx

```diff
@@ -0,0 +1,382 @@
+        value: this.visibleRows.length,
+      },
+      concepts: {
+        name: "Concepts", displayOrder: displayOrder++,
+        value: concept_ids.length,
+        hiddenConceptCnt: cidDiff(concept_ids, visibleCids),
+      },
+      expressionItems: {
+        name: "Definition concepts", displayOrder: displayOrder++,
+        value: this.specialConcepts.expressionItems.length,
+        hiddenConceptCnt: cidDiff(this.specialConcepts.expressionItems, visibleCids),
+        hideByDefault: false,
+      },
+      added: {
+        name: "Added", displayOrder: displayOrder++,
+        value: get(this.specialConcepts, '.added.length', undefined),
+        hiddenConceptCnt: cidDiff(this.specialConcepts.added, visibleCids),
+        hideByDefault: true,
+      },
+      removed: {
+        name: "Removed", displayOrder: displayOrder++,
+        value: get(this.specialConcepts, '.removed.length', undefined),
+        hiddenConceptCnt: cidDiff(this.specialConcepts.removed, visibleCids),
+        hideByDefault: true,
+      },
+      expansion: {
+        name: "Expansion concepts", displayOrder: displayOrder++,
+        value: uniq(flatten(Object.values(csmi).map(Object.values))
+                .filter(c => c.csm).map(c => c.concept_id)).length,
+        hiddenConceptCnt: cidDiff(concept_ids, visibleCids),
+      },
+      standard: {
+        name: "Standard concepts", displayOrder: displayOrder++,
+        value: concepts.filter(c => c.standard_concept === 'S').length,
+      },
+      classification: {
+        name: "Classification concepts", displayOrder: displayOrder++,
+        value: concepts.filter(c => c.standard_concept === 'C').length,
+      },
+      nonStandard: {
+        name: "Non-standard", displayOrder: displayOrder++,
+        value: this.specialConcepts.nonStandard.length,
+        hiddenConceptCnt: cidDiff(this.specialConcepts.nonStandard, visibleCids),
+        hideByDefault: false,
+      },
+    }
+    for (let type in rows) {
+      let row = {...get(this, ['statsOptionsRows', type], {}), ...rows[type]};
+      if (typeof(row.value) === 'undefined') {
+        delete rows[type];
+        continue;
+      }
+      row.type = type;
+      if (isEmpty(this.statsOptionsRows) && typeof(row.hideByDefault) !== 'undefined') {
+        row.isHidden = row.hideByDefault;
+      }
+      rows[type] = row;
+    }
+    this.statsOptionsRows = rows;
+  };
+  getStatsOptionsRows() {
+    return sortBy(this.statsOptionsRows, d => d.displayOrder);
+  }
+
+  addNodeToVisible(nodeId, displayedRows, alwaysShow, depth = 0) {
+    const node = {...this.nodes[nodeId], depth};
+    displayedRows.push(node);
+    const childIds = this.graph.outNeighbors(nodeId); // Get outgoing neighbors (children)
+    if (node.expanded) {
+      childIds.forEach(childId => {
+        this.addNodeToVisible(childId, displayedRows, alwaysShow, depth + 1); // Recurse
+      });
+    } else {
+      alwaysShow.forEach(alwaysShowId => {
```

Comment:
Siggie [wrote](https://github.com/jhu-bids/TermHub/commit/666dd307e66ab1ddf4ce18bfd2e80f957f5d98e6):
 Changes to alwaysShow/specialConcepts/options...; working better but
  still at least one bug (showing always show higher up in tree than it
  should)
  - Need to document that code. Maybe can do code review with
    @joeflack4. It's pretty gnarly because of getting data at different
    points and needing to rerender on gcDispatch TOGGLE_OPTION, and
    weirdness with using mutable objects in reducer and stuff.
  - And need to treat things to hide differently from things to always
    show.
- Can now close FlexibleContainers with escape key. Had to fix some
  stuff with zIndexes. It's confusing code but seems to work now.
- Tried to shorten long cset names with css text-overflow: elipsis, but
  can't get it to work.